providers/cassandra/cassandra.go

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cassandra
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/erda-project/erda-infra/base/logs"
+	"github.com/erda-project/erda-infra/base/servicehub"
+	writer "github.com/erda-project/erda-infra/pkg/parallel-writer"
+	"github.com/gocql/gocql"
+)
+
+// WriterConfig .
+type WriterConfig struct {
+	Parallelism uint64 `file:"parallelism" default:"4" desc:"parallelism"`
+	Batch       struct {
+		Size    uint64        `file:"size" default:"100" desc:"batch size"`
+		Timeout time.Duration `file:"timeout" default:"30s" desc:"timeout to flush buffer for batch write"`
+	} `file:"batch"`
+	Retry int `file:"retry" desc:"retry if fail to write"`
+}
+
+// SessionConfig .
+type SessionConfig struct {
+	Keyspace    KeyspaceConfig `file:"keyspace"`
+	Consistency string         `file:"consistency" default:"LOCAL_ONE"`
+}
+
+// KeyspaceConfig .
+type KeyspaceConfig struct {
+	Name        string                    `file:"name" env:"CASSANDRA_KEYSPACE"`
+	Auto        bool                      `file:"auto"`
+	Replication KeyspaceReplicationConfig `file:"replication"`
+}
+
+// KeyspaceReplicationConfig .
+type KeyspaceReplicationConfig struct {
+	Class  string `file:"class" default:"SimpleStrategy"`
+	Factor int32  `file:"factor" default:"2"`
+}
+
+// Cassandra .
+type Cassandra interface {
+	CreateKeyspaces(ksc ...*KeyspaceConfig) error
+	Session(cfg *SessionConfig) (*gocql.Session, error)
+	NewBatchWriter(session *gocql.Session, c *WriterConfig, builderCreator func() StatementBuilder) writer.Writer
+}
+
+type config struct {
+	Hosts    string        `file:"host" env:"CASSANDRA_ADDR" default:"localhost:9042" desc:"server hosts"`
+	Security bool          `file:"security" env:"CASSANDRA_SECURITY_ENABLE" default:"false" desc:"security"`
+	Username string        `file:"username" env:"CASSANDRA_SECURITY_USERNAME" default:"" desc:"username"`
+	Password string        `file:"password" env:"CASSANDRA_SECURITY_PASSWORD" default:"" desc:"password"`
+	Timeout  time.Duration `file:"timeout" env:"CASSANDRA_TIMEOUT" default:"3s" desc:"session timeout"`
+}
+
+type define struct{}
+
+func (d *define) Service() []string   { return []string{"cassandra"} }
+func (d *define) Summary() string     { return "cassandra" }
```

Comment:
Use `Services` to describe dependencies, not `Service`

Reply:
We may need a contributing guideline